File: Java/02.java
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class Anagrama {
    /*
     * Escribe una función que reciba dos palabras (String) y retorne
     * verdadero o falso (Bool) según sean o no anagramas.
     * - Un Anagrama consiste en formar una palabra reordenando TODAS
     *   las letras de otra palabra inicial.
     * - NO hace falta comprobar que ambas palabras existan.
     * - Dos palabras exactamente iguales no son anagrama.
     */
    public Boolean isAnagram(String p1, String p2){
        p1 = p1.toLowerCase().replaceAll(" ", "");
        p2 = p2.toLowerCase().replaceAll(" ", "");
        if (p1.equals(p2) || p1.length() != p2.length()){
            return false;
        }
        Map<Character, Integer> wordsp1 = new HashMap<>();
        Map<Character, Integer> wordsp2 = new HashMap<>();
        for (int i = 0; i < p1.length(); i++) {
            int valorActualp1 = wordsp1.getOrDefault(p1.charAt(i), 0);
            wordsp1.put(p1.charAt(i), valorActualp1 + 1);
        }
        for (int i = 0; i < p2.length(); i++) {
            int valorActualp2 = wordsp2.getOrDefault(p2.charAt(i), 0);
            wordsp2.put(p2.charAt(i), valorActualp2 + 1);
        }
        for (Character ch : wordsp1.keySet()){
            if (!wordsp2.containsKey(ch)){
                return false;
            }
            if (!Objects.equals(wordsp1.get(ch), wordsp2.get(ch))){
                return false;
            }
        }
        return true;
    }
}

File: C#/02.cs
/*
* Escribe una función que reciba dos palabras (String) y retorne
* verdadero o falso (Bool) según sean o no anagramas.
* - Un Anagrama consiste en formar una palabra reordenando TODAS
*   las letras de otra palabra inicial.
* - NO hace falta comprobar que ambas palabras existan.
* - Dos palabras exactamente iguales no son anagrama.
*/
   public class Anagrama
   {
       public bool esAnagrama(string p1, string p2)
       {
           p1 = p1.ToLower().Trim();
           p2 = p2.ToLower().Trim();
           if (p1.Equals(p2)) return false;
           if (p1.Length != p2.Length) return false;
           Dictionary<char, int> mapp1 = new Dictionary<char, int>();
           Dictionary<char, int> mapp2 = new Dictionary<char, int>();
           for (int i = 0; i < p1.Length; i++)
           {
               if (!mapp1.ContainsKey(p1[i]))
               {
                   mapp1.Add(p1[i], 1);
               }
               else
               {
                   mapp1[p1[i]] = mapp1[p1[i]] + 1;
               }
           }
           for (int i = 0; i < p2.Length; i++)
           {
               if (!mapp2.ContainsKey(p2[i]))
               {
                   mapp2.Add(p2[i], 1);
               }
               else
               {
                   mapp2[p2[i]] = mapp2[p2[i]] + 1;
               }
           }
           foreach (char character in mapp1.Keys)
           {
               if (!mapp2.ContainsKey(character)) return false;
               if (mapp1[character] != mapp2[character]) return false;
           }
           return true;
       }
   }